### PR TITLE
[AIRFLOW-2895] Prevent scheduler from spamming heartbeats/logs

### PR DESCRIPTION
**Pull request by aoen**
Reverts most of AIRFLOW-2027 until the issues with it can be fixed.

### Jira

- [X] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-2895

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Reverting some broken code that is missing tests.

### Commits

- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [X] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [X] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

@bolkedebruin

### CODE REVIEW

File: UPDATING.md [hunk 1]
```diff
@@ -420,10 +420,6 @@ indefinitely. This is only available on the command line.
 
 After how much time should an updated DAG be picked up from the filesystem.
 
-#### min_file_parsing_loop_time
```
Comment (feng-tao) [inline UPDATING.md]: do we need to update the UPDATING.MD given we remove this variable from the config file?

Comment (aoen) [reply UPDATING.md]: Yeah now that 1.10 release has gone out, you are right, thanks. I added it back and made a note that it's disabled.

File: UPDATING.md [hunk 2]
```diff
@@ -421,7 +421,7 @@ indefinitely. This is only available on the command line.
 After how much time should an updated DAG be picked up from the filesystem.
 
 #### min_file_parsing_loop_time
-
+CURRENTLY DISABLED DUE TO A BUG
```
Comment (bolkedebruin) [inline UPDATING.md:436]: Please add a note to the master section, we are not changing the past. People will read it as “I am at this version, what do I need to do to get to the next”

Comment (aoen) [reply UPDATING.md]: Done

Comment (aoen) [reply UPDATING.md:436]: Done, PTAL

File: UPDATING.md (hunk 1 above)
Comment (aoen) [reply UPDATING.md]: @feng-tao done, PTAL

**Comment by feng-tao** [comment]
hey @aoen, lgtm, and based on the original pr(https://github.com/apache/incubator-airflow/pull/2986), do you need to revert the tests as well?

**Comment by aoen** [comment]
Thank you! I kept the test changes because they were separate improvements/fixes (e.g. an unused variable).

**Comment by cinhil** [comment]
Hello,
Do you intend to merge it (backport) on v1-10-stable branch ?
Maybe aiming an 1.10.1 later... This issue block us to push 1.10 in production.

**Comment by bolkedebruin** [comment]
1.10.1

**Comment by ashb** [comment]
Thanks, I'm starting to make a note of what PRs we'd want to get in to a 1.10.1 release.

**Comment by Fokko** [comment]
Thanks @ashb 

My suggestion would be to branch of the 1.10 branch and cherry-pick some commits into this branch. Would also be awesome to make it compatible with Python 3.7. Cheers!

**Pull request closed without merging**